### DISCUSSION TITLE
Lowering minimal required cmake version

### DISCUSSION THREAD

**Discussion by dev-0x7C6** [gabime/spdlog#2215] [Q&A]
Hi,

I'm wondering is 3.10 cmake actual minimum required version. I used spdlog in my projects and I have some build environments when unfortunately cmake is pretty old (like 3.5.x). Do you have any advice how to progress with that? Is spdlog already tuned to run with lowest cmake (without sacrifice in cmake features/syntax)?

Regards,
Bartłomiej

**Comment by dev-0x7C6** [comment]
I found that `fmt` that I using as external library is more forgiving on minimal cmake. i. e. `cmake_minimum_required(VERSION 3.1...3.18)`

**Reply by rroguski** [reply]
From cmake docs for`cmake_minimum_required` (note the bolded text):

> The cmake_minimum_required(VERSION) command implicitly invokes the cmake_policy(VERSION)
> command to specify that the current project code is written for the given range of CMake versions.
> All policies known to the running version of CMake and introduced in the <min> (or <max>, if specified)
> version or earlier will be set to use NEW behavior. **All policies introduced in later versions will be unset.**
> This effectively requests behavior preferred as of a given CMake version and tells newer CMake versions
> to warn about their new policies.

Which means that setting a different version in `cmake_minimum_required` makes cmake run in that version regardless of the installed version. 
I just cloned the repo and changed the minimum version to 3.4 and cmake ran fine.

But that requires making changes in `spdlog` which might not always be possible (I use `spdlog` as a submodule, with each clone I'd have to manually change the version).

The best thing to do in my opinion is download a binary distribution from the [cmake downloads page](https://cmake.org/download/) on your system with an older cmake version. If you're on linux I recommend using [stow](https://linux.die.net/man/8/stow) for installing precompiled binaries using symlinks.

**Reply by dev-0x7C6** [reply]
Got it, since I'm not dealing with host system but with an old embedded system build environment and sdks, I was forced to patch spdlog that will decrease minimal version to 3.4. It looks like it's going to work fine.

Thanks for pinpointing 3.4.
